src/Rap2hpoutre/LaravelLogViewer/Pattern.php

```diff
@@ -4,44 +4,50 @@
 
 /**
  * Class Pattern
- * @property array patterns
  * @package Rap2hpoutre\LaravelLogViewer
  */
 
 class Pattern
 {
 
     /**
-     * @var array
+     * @return array
      */
-    private $patterns = [
-        'logs' => '/\[\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}([\+-]\d{4})?\].*/',
-        'current_log' => [
-            '/^\[(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}([\+-]\d{4})?)\](?:.*?(\w+)\.|.*?)',
-            ': (.*?)( in .*?:[0-9]+)?$/i'
-        ],
-        'files' => '/\{.*?\,.*?\}/i',
-    ];
+    public function all()
+    {
+        return array_keys($this->patterns());
+    }
 
     /**
-     * @return array
+     * @param array $options Override default `date` and/or `loglevels` regexes.
+     * @return string[]
      */
-    public function all()
+    protected function patterns($options = [])
     {
-        return array_keys($this->patterns);
+        $options = array_merge([
+            'date' => '\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}(?:[\+-]\d{4})?',
+            'loglevels' => '',
+        ], $options);
+
+        return [
+            // Separate log file into log entries.
+            'logs' => '/\[' . $options['date'] . '\].*(?:\R(?!\[' . $options['date'] . '\]).*)*/',
+
+            // Capture 1: date, 2: context, 3: loglevel 4: message, 5: file.
+            'heading' => '/^\[('. $options['date'] . ')\](?:.*?(\w+)\.|.*?)(' . $options['loglevels'] . '): (.*?)( in .*?:[0-9]+)?$/i',
+
+            'files' => '/\{.*?\,.*?\}/i',
+        ];
     }
 
     /**
      * @param $pattern
-     * @param null $position
+     * @param array $options
      * @return string pattern
      */
-    public function getPattern($pattern, $position = null)
+    public function getPattern($pattern, $options = [])
```

Comment:
added flexibility options to allow regex modification by passing explicit loglevels or date-syntax.